### PR TITLE
docs: adding system test overview

### PR DESCRIPTION
**Pull request by danlavu**

### CODE REVIEW

**Comment by pbrezina** [comment]
Hi Dan, LGTM

File: src/contrib/running-tests.rst [hunk 1]
```diff
@@ -1,24 +1,33 @@
 Running Tests
 =============
 
-SSSD is a complex piece of software with a long development history. Therefore, there are several layers of tests with different goals and using different frameworks. This page shows how to run the tests and how to add new ones or modify the existing tests.
+SSSD is a complex piece of software with a long development history. Therefore, there has been several layers of tests with different goals and using different frameworks. We are currently in progress of reworking all of our tests and standardizing them using our own `SSSD Test Framework <https://tests.sssd.io/en/latest/>`_.
 
-Existing test types
--------------------
+To get started immediately you can visit `Running Tests <https://tests .sssd.io/en/latest/running-tests.html>`_ page in the `SSSD Test Framework <https://tests.sssd.io/en/latest/>`_ wiki and start following the instructions provided.
```
Comment (justin-stephenson) [inline src/contrib/running-tests.rst:6]: `https://tests .sssd.io/en/latest/running-tests.html`

Looks like an extra space after `tests` is here.

File: src/contrib/running-tests.rst [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
 
-Each test is different. Sometimes, you want to test the whole system end-to-end, sometimes the test should exercise some corner case for which special input and environment must be simulated. This section should give you a better idea what kind of tests already exist in SSSD so that you can choose where to add a new test and also provides a general overview.
+Overview
+--------
+
+Our system tests are written using Pytest and `Pytest Multihost Plug-in <https://github.com/next-actions/pytest-mh>`_ orchestrating the setup, management, execution and teardown of hosts and tests.
+
+All major identity providers; LDAP, Kerberos, FreeIPA, Microsoft Active Directory, Samba Active Directory are provisioned as VMs or are containers and are intended to be provisioned on each run, ensuring the machines are in a clean state. The setup code is in our `SSSD CI Containers Repository <https://github.com/sssd/sssd-ci-containers/>`_. Note, these machines are not to be used for production and should only be used in development and local testing only.
```
Comment (justin-stephenson) [inline src/contrib/running-tests.rst:13]: s/or are containers/or containers/

File: src/contrib/running-tests.rst [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
+
+For a comprehensive test run, will require several hosts; client, nfs, dns, ipa, krb, samba, ldap and ad. Emulating several environments and scenarios needed by the tests. In local testing, all containers use podman for containerization then Vagrant and KVM for virtualization. Two networks are created for each virtual computing service and needs to be routable to one another. For simplicity, all documentation assumes that one physical host is used to run the tests.
```
Comment (justin-stephenson) [inline src/contrib/running-tests.rst:15]: s/For a comprehensive test run, will require several hosts/A comprehensive test run requires several hosts/

File: src/contrib/running-tests.rst [hunk 4]
```diff
@@ -1,24 +1,33 @@
...
+
+The bulk of the tests are of a generic provider and will run against all supported topologies, but some tests are written to be run against a specific topology or topologies. Topologies also define which hosts are needed by the test. For example, AD topology will require a client host and an AD host, Samba topology will require a client and a Samba host but both can be used by the GenericAD topology group will run against both AD and Samba.
```
Comment (justin-stephenson) [inline src/contrib/running-tests.rst:17]: `The bulk of the tests are of a generic provider` sounds like poor english IMO

Maybe `the bulk of tests are written to utilize a generic provider topology` ?

File: src/contrib/running-tests.rst [hunk 5]
```diff
@@ -1,24 +1,33 @@
...
+
+Tests are also written so it can be ran independently, any test setup that is needed by the test, is executed right before the test code and is undone or the host is refreshed for the following test. For example, 389 as an LDAP provider is destroyed and re-installed before every test that, while AD the setup is undone.
```
Comment (justin-stephenson) [inline src/contrib/running-tests.rst:19]: s/Tests are also written so it can be ran independently/Tests are also written so they can be ran independently/

File: src/contrib/running-tests.rst [hunk 6]
```diff
@@ -1,24 +1,33 @@
...
+
+Test Types
+----------
+
+Historically we have had four different test types on using different frameworks. To get started, please reference system tests, because we will be retire-ring the integration and multihost tests.
```
Comment (justin-stephenson) [inline src/contrib/running-tests.rst:24]: s/using different frameworks/different frameworks/

File: src/contrib/running-tests.rst (hunk 4 above)
Comment (danlavu) [reply src/contrib/running-tests.rst]: "The tests are designed to be generic, and will run against all supported topologies, " ?
